src/IoC/InstancePool.cs

```diff
@@ -0,0 +1,50 @@
+﻿namespace IoC;
+
+public sealed class InstancePool
+{
+    //This is where the instances will be stored.
+    private static Dictionary<string, object>? _implementationStore;
+    //Make the contructor unaccessible.
+    InstancePool()
+    {
+        _implementationStore = new Dictionary<string, object>();
+    }
+
+
+    //This is the only way to access to the methods of the class.
+    private static InstancePool? _instance = null;
```

Comment:
Same here

Reply:
The problem with this field is when is marked as `readonly` the compiler is expecting to set the value via constructor in the `InstancePool` class. So that will break the `singleton pattern`.

Reply:
I see makes sense